zephyr/src/logging/spotflow_log_backend.c

```diff
@@ -0,0 +1,193 @@
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/mqtt.h>
+#include <zephyr/kernel.h>
+
+#include "spotflow_cbor.h"
+#include "net/spotflow_processor.h"
+#include "spotflow_cbor_output_context.h"
+
+LOG_MODULE_REGISTER(log_backend_spotflow, CONFIG_SPOTFLOW_PROCESSING_BACKEND_LOG_LEVEL);
+
+struct spotflow_log_context {
+	struct spotflow_cbor_output_context* cbor_output_context;
+	size_t dropped_backend_count;
+	size_t message_index;
+};
+
+static struct spotflow_log_context spotflow_log_ctx;
+
+static void init(const struct log_backend* backend);
+
+static void process(const struct log_backend* backend, union log_msg_generic* msg);
+
+static void panic(const struct log_backend* backend);
+
+static void dropped(const struct log_backend* backend, uint32_t cnt);
+
+static const struct log_backend_api log_backend_spotflow_api = {
+	.init = init,
+	.process = process,
+	.panic = panic,
+	.dropped = dropped,
+	/* .sync_string and .sync_hexdump can be added if you use IMMEDIATE mode */
+};
+
+#ifdef CONFIG_LOG_BACKEND_SPOTFLOW
+/* can be autostarted because there is message queue where messages are stored */
+LOG_BACKEND_DEFINE(log_backend_spotflow, log_backend_spotflow_api, true /* autostart */,
+		   &spotflow_log_ctx);
+#endif /* CONFIG_LOG_BACKEND_SPOTFLOW */
+
+static int enqueue_log_msg(const struct spotflow_mqtt_msg* msg, struct spotflow_log_context* ctx);
+
+static void process_single_message_stats_update(struct spotflow_log_context* context, bool dropped);
+
+static void process_message_stats_update(struct spotflow_log_context* context, uint32_t cnt,
+					 bool dropped);
+
+static void init(const struct log_backend* const backend)
+{
+	LOG_INF("Spotflow log backend initialized.");
+	__ASSERT(backend->cb->ctx != NULL, "Spotflow log backend context is NULL");
+	struct spotflow_log_context* ctx = backend->cb->ctx;
+	int rc = spotflow_cbor_output_context_init(&ctx->cbor_output_context);
+	__ASSERT(result == 0, "Failed to initialize spotflow output context");
+	if (rc < 0) {
+		LOG_ERR("Failed to initialize spotflow output context: %d", rc);
+		return;
+	}
+	ctx->dropped_backend_count = 0;
+	ctx->message_index = 0;
+	spotflow_start_mqtt();
```

Comment:
```suggestion
	__ASSERT(backend->cb->ctx != NULL, "Spotflow log backend context is NULL");
	struct spotflow_log_context* ctx = backend->cb->ctx;
	int rc = spotflow_cbor_output_context_init(&ctx->cbor_output_context);
	if (rc < 0) {
		LOG_ERR("Failed to initialize spotflow output context: %d", rc);
		return;
	}
	ctx->dropped_backend_count = 0;
	ctx->message_index = 0;

	spotflow_start_mqtt();

	LOG_INF("Spotflow logging backend initialized.");
```

Maybe also add `Initializing Spotflow logging backend.` at the start?

Reply:
maybe as debug only?